src/transformers/generation/logits_process.py

```diff
@@ -981,3 +981,65 @@ def __call__(self, input_ids, scores):
                 scores[k, : self.timestamp_begin] = -float("inf")
 
         return scores
+
+
+class SemanticLogitsProcessor(LogitsProcessor):
+    r"""
+    [`LogitsProcessor`] enforcing that logits from the semantic model observe Bark original logic.
```

Comment:
This is not a great docstring - we should describe _what_ the logits processor does, rather than say it does 'Bark behaviour'. Also now that we're in the logits processor file, these logits processors should be general across models (as much as possible!)